### PR TITLE
VendorConfiguration: speed up collecting definition lines

### PR DESCRIPTION
**Pull request by dhalperi**
All these streams are slow.

### CODE REVIEW

File: projects/batfish-common-protocol/src/main/java/org/batfish/vendor/VendorConfiguration.java [hunk 1]
```diff
@@ -221,26 +220,23 @@ public void undefined(StructureType structureType, String name, StructureUsage u
...
-                    Optional.ofNullable(extraLines)
-                        .map(el -> el.get(originalLine))
-                        .map(set -> set.stream().mapToInt(Integer::intValue))
-                        .orElse(IntStream.of()));
-              } else if (child instanceof RuleContext) {
-                return collectLines((RuleContext) child, parser, extraLines);
-              }
-              return IntStream.empty();
-            })
-        .distinct();
```
Comment (dhalperi) [inline projects/batfish-common-protocol/src/main/java/org/batfish/vendor/VendorConfiguration.java:243]: Q: how much does this `.distinct()` matter. Having a hard time finding a benchmark that would have invoked it.

**Review by progwriter — COMMENTED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/8933)* status: all files reviewed, 1 unresolved discussion (waiting on @dhalperi)

___
*[`projects/batfish-common-protocol/src/main/java/org/batfish/vendor/VendorConfiguration.java` line 0 at r1](https://reviewable.io/reviews/batfish/batfish/8933#gh-1462625356:-NonxEoCBnbPa_o_8mky:b-i252wh) ([raw file](https://github.com/batfish/batfish/blob/e300a56401dd84cfbd0693f1773be7a119aae4ba/projects/batfish-common-protocol/src/main/java/org/batfish/vendor/VendorConfiguration.java#L0)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map [this GitHub inline comment thread](https://github.com/batfish/batfish/pull/8933#discussion_r1462625356) to the right spot — sorry!)*

Q: how much does this `.distinct()` matter. Having a hard time finding a benchmark that would have invoked it.
</blockquote></details>

I was going to ask the same question :)

**Review by dhalperi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/8933)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @dhalperi)

___
*[`projects/batfish-common-protocol/src/main/java/org/batfish/vendor/VendorConfiguration.java` line 0 at r1](https://reviewable.io/reviews/batfish/batfish/8933#gh-1462625356:-NonyiOy7-VJvVKqqO1T:b-qzc956) ([raw file](https://github.com/batfish/batfish/blob/e300a56401dd84cfbd0693f1773be7a119aae4ba/projects/batfish-common-protocol/src/main/java/org/batfish/vendor/VendorConfiguration.java#L0)):*
<details><summary><i>Previously, progwriter (Victor Heorhiadi) wrote…</i></summary><blockquote>

I was going to ask the same question :)
</blockquote></details>

https://github.com/batfish/batfish/pull/6249/files says I introduced this, so I say not much

**Review by dhalperi — COMMENTED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/batfish/batfish/8933)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @dhalperi)

**Pull request merged**